pkg/distributor/distributor.go

```diff
@@ -952,7 +952,12 @@ func (d *Distributor) prePushValidationMiddleware(next PushFunc) PushFunc {
 			d.discardedSamplesRateLimited.WithLabelValues(userID, group).Add(float64(validatedSamples))
 			d.discardedExemplarsRateLimited.WithLabelValues(userID).Add(float64(validatedExemplars))
 			d.discardedMetadataRateLimited.WithLabelValues(userID).Add(float64(validatedMetadata))
-			return newIngestionRateLimitedError(d.limits.IngestionRate(userID), d.limits.IngestionBurstSize(userID))
+
+			burstSize := d.limits.IngestionBurstSize(userID)
+			if d.limits.IngestionBurstFactor(userID) > 0 {
+				burstSize = int(d.limits.IngestionRate(userID))*d.limits.IngestionBurstFactor(userID)
+			}
```

Comment:
Have you considered just moving this logic to `IngestionBurstSize()`? Basically the `IngestionBurstSize()` can just return the burst size to apply, which could be the one configured or the new one based on the factor.

Reply:
I'm not sure if we can do that because we divide the burst factor result by the number of distributors and we do not do that with the burst size in the rate limit strategy.